Sofa/framework/Helper/src/sofa/helper/system/FileSystem.h

```diff
@@ -93,12 +93,31 @@ static bool removeDirectory(const std::string& path);
 /// @return true on error
 static bool removeAll(const std::string& path) ;
 
+/// @brief The file or empty directory identified by the path is deleted
+///
+/// @return true if the file was deleted, false if it did not exist.
 static bool removeFile(const std::string& path);
 
 /// @brief check that all element in the path exists or create them. (This function accepts relative paths)
 ///
 /// @return the valid path.
-static std::string findOrCreateAValidPath(const std::string path) ;
+SOFA_HELPER_FILESYSTEM_FINDORCREATEAVALIDPATH_DEPRECATED()
+static std::string findOrCreateAValidPath(const std::string path);
+
+/// @brief Ensures that a folder exists at the specified path. If the folder does not exist, it will be created.
+///
+/// @param pathToFolder The path of the folder to ensure exists.
+///
+/// @note The function assumes that a path to a folder is given.
+static void ensureFolderExists(const std::string& pathToFolder);
+
+/// @brief Ensures that the folder containing the specified file path exists. If the folder does not
+/// exist, it will be created. If the file does not exist, it will not be created.
+///
+/// @param pathToFile The path to the file. The function ensures that the directory containing this file exists.
+///
+/// @note The function assumes that a path to a file is given.
+static void ensureFolderForFileExists(const std::string& pathToFile);
```

Comment:
why just not, "createFolder" and if folder already exists, it does nothing?

Reply:
this is what `ensureFolderForFileExists` does, but recursively

Reply:
yes, I was just speaking about the name of the function that I find a bit complex.

Reply:
lol too late anyway... @hugtalbot already merged it 😆